tests/nat/conftest.py

```diff
@@ -0,0 +1,227 @@
+import os
+import copy
+
+import pytest
+
+from nat_helpers import SUPPORTED_TOPO
+from nat_helpers import SETUP_CONF
+from nat_helpers import DUT_TMP_DIR
+from nat_helpers import NAT_ADMIN_MODE
+from nat_helpers import TEMPLATE_DIR
+from nat_helpers import GLOBAL_NAT_TIMEOUT
+from nat_helpers import GLOBAL_TCP_NAPT_TIMEOUT
+from nat_helpers import GLOBAL_UDP_NAPT_TIMEOUT
+from nat_helpers import NAT_GLOBAL_TEMPLATE
+from nat_helpers import conf_ptf_interfaces
+from nat_helpers import teardown_test_env
+from nat_helpers import exec_command
+from nat_helpers import dut_interface_control
+from common.config_reload import config_reload
+
+
+@pytest.fixture(params=['TCP', 'UDP', 'ICMP'])
+def protocol_type(request):
+    """
+    used to parametrized test cases on protocol type
+    :param request: pytest request object
+    :return: protocol type
+    """
+    return request.param
+
+
+@pytest.fixture(params=['TCP', 'UDP'])
+def protocol_type_no_icmp(request):
+    """
+    used to parametrized test cases on protocol type
+    :param request: pytest request object
+    :return: protocol type
+    """
+    return request.param
+
+
+@pytest.fixture(autouse=True)
+def teardown(duthost):
+    """
+    Teardown procedure for all test function
+    :param duthost: DUT host object
+    """
+    yield
+    # Teardown after test finished
+    shutdown_cmds = ["config nat remove {}".format(cmd) for cmd in ["static all", "bindings", "pools", "interfaces"]]
+    exec_command(duthost, shutdown_cmds)
+    # Clear all enries
+    duthost.command("sonic-clear nat translations")
+
+
+@pytest.fixture(params=["loopback", "port_in_lag"], scope='module')
+def setup_test_env(request, ptfhost, duthost, testbed):
+    """
+    setup testbed's environment for CT run
+    :param request: pytest request object
+    :param ptfhost: PTF host object
+    :param duthost: DUT host object
+    :param testbed: Testbed object
+    :return: interface type and testbed setup info
+    """
+    # Check if topology is supported
+    if testbed['topo']['name'] not in SUPPORTED_TOPO:
+        pytest.skip('Unsupported topology')
+    interface_type = request.param
+    interfaces_nat_zone = {}
+    portchannels_port_indices = []
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="persistent")['ansible_facts']
+    router_mac = duthost.setup()['ansible_facts']['ansible_Ethernet0']['macaddress']
+    config_port_indices = cfg_facts['port_index_map']
```

Comment:
i think the router mac is available in duthost fixture

Reply:
Good  catch, will be updated with next commit